components/mpas-framework/src/framework/mpas_dmpar.F

```diff
@@ -8832,23 +8832,30 @@ subroutine mpas_dmpar_exch_group_pack_buffer_field3d_real(exchangeGroup, field,
          bufferOffset = commListPtr % bufferOffset
          nAdded = 0
          fieldCursor => field
+         call mpas_log_write('exchange field'//trim(fieldCursor % fieldName))
          do while ( associated(fieldCursor) )
+           call mpas_log_write('point to exchList')
            exchListPtr => fieldCursor % sendList % halos(haloLayer) % exchList
            do while ( associated(exchListPtr) )
              if ( exchListPtr % endPointID == commListPtr % procID ) then
                do iExch = 1, exchListPtr % nList
+                 call mpas_log_write('iter on nList, '//trim(fieldCursor % dimNames(1))//trim(fieldCursor % dimNames(2)))
                  do j = 1, fieldCursor % dimSizes(2)
                    do k = 1, fieldCursor % dimSizes(1)
+                     call mpas_log_write('get iBuffer')
                      iBuffer = (exchListPtr % destList(iExch) - 1) * fieldCursor % dimSizes(1) * fieldCursor % dimSizes(2) &
                                + (j - 1) * fieldCursor % dimSizes(1) + k + bufferOffset
+                     call mpas_log_write('get array')
                      commListPtr % rbuffer(iBuffer) = fieldCursor % array(k, j, exchListPtr % srcList(iExch))
```

Comment:
@akturner or others, I'm on holiday next week. As I mentioned on slack, the 2d mask on a 2d var appears to be working (at the commit https://github.com/E3SM-Ocean-Discussion/E3SM/pull/25/commits/345440573f7f72a2183efee3e0b60ef3da049603). I added some of the infrastructure for adding a 2d mask to the 3d vararray activeTracers. However, there is a seg fault at this line that I didn't get a chance to track down. Suggestions welcome!